include/nvexec/detail/config.cuh

```diff
@@ -15,8 +15,16 @@
  */
 #pragma once
 
+#include "../../stdexec/__detail/__config.hpp"
+
 #if !defined(_NVHPC_CUDA) && !defined(__CUDACC__)
 #error The NVIDIA schedulers and utilities require CUDA support
 #endif
 
 #define STDEXEC_STREAM_DETAIL_NS _strm
+
+#if STDEXEC_CLANG()
+#define STDEXEC_CLANG_HOST_DEVICE __host__ __device__
```

Comment:
Why "CLANG"? We'll need this eventually for nvcc, right?

Reply:
True. What would be a good name for a CUDA thing that isn't needed with nvc++? I guess I could call it `STDEXEC_CUDACC_HOST_DEVICE`.

Reply:
I think `STDEXEC_HOST_DEVICE` would be fine and would be canonical with what most other libraries do.

Reply:
The reason why is that according to @senior-zero, there are some `__host__ __device__` attributes that nvc++ needs, but others that would cause it problems. @senior-zero can you chime in here?

Reply:
The idea is that we need implicit attributes here. Things like `visit` can be used on both cpu and gpu, but function that are provided to `visit` are cpu or gpu only, not mixed ones. The origin of `__host__ __device__` here is only to fix clangd, I don't want to mark these functions as `__host__ __device__` in the nvhpc builds yet, so the macro should do nothing for nvc++. Once we start working on nvcc, we'll have to revisit this question and probably dispatch within the function using `NV_IF_TARGET`. Also, I wouldn't like the macro to be the part of the API, so maybe `STDEXEC_DETAIL_*` prefix would help as well.